src/WebJobs.Script.WebHost/Diagnostics/FunctionsVersionLogger.cs

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Azure.WebJobs.Hosting;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Script.Diagnostics
+{
+    /// <summary>
+    /// Service responsible for logging versions
+    /// </summary>
+    public class FunctionsVersionLogger : IHostedService, IDisposable
+    {
+        private readonly ILogger _logger;
+        private readonly Timer _timer;
+        private readonly TimeSpan _interval;
+        private readonly IEnvironment _environment;
+        private bool _disposed;
+
+        public FunctionsVersionLogger(ILoggerFactory loggerFactory, IEnvironment environment)
+        {
+            _logger = loggerFactory.CreateLogger<FunctionsVersionLogger>();
+            _interval = TimeSpan.FromMinutes(10);
```

Comment:
I thought this was supposed to only log every hour?

Reply:
Updated